### PR TITLE
Removal of Node 6

### LINKED ISSUE #700: Remove Node v6
(To be addresses in May 2019)

Node v6 ends support in April 2019 (see https://github.com/nodejs/Release) and should be removed by that date. In particular, this issue involves the following actions:

* Node v6 should be removed from .travis.yml
* engine version should be modified to be >=8 in packages.json

### PR DESCRIPTION
**Pull request by jason-fox**
Fixes #700 

*  Remove 6 from travis - add 12
*  Update minimum node engine to 8
*  Update dependencies to latest Node 8

### CODE REVIEW

**Comment by jason-fox** [comment]
@Jagatjot @fgalan - Please review

File: package.json [hunk 1]
```diff
@@ -38,31 +38,31 @@
     "watch": "watch 'npm test && npm run lint' ./lib ./test"
   },
   "dependencies": {
```
Comment (fgalan) [inline package.json]: Are the dependencies and devDependencies changed motivated by the node 12 addition? In other words, if the modules are not upgraded this way node 12 job in travis fails?

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 language: node_js
```
Comment (fgalan) [inline .travis.yml]: A line in CHANGES_NEXT_RELEASE about the removal of Node 6 should be include. For homogeneity, I'd suggest to use the same in https://github.com/telefonicaid/iotagent-json/pull/400:

> Set Nodejs 8 as minimum version in packages.json (effectively removing Nodev6 from supported versions)

File: package.json (hunk 1 above)
Comment (jason-fox) [reply package.json]: No, as summarized in the intro, they have been updated to the latest **supported** version for node 8 - this closes several outstanding vulnerabilities which were present due to using outdated version. 

I had assumed that the older versions were being kept for **node 6** support.

File: .travis.yml (hunk 2 above)
Comment (jason-fox) [reply .travis.yml]: Fixed: ab10c3b

File: package.json (hunk 1 above)
Comment (fgalan) [reply package.json]: Please add the changes to CHANGES_NEXT_RELEASE in this style:

```
- Upgrade from node-uuid ~1.4.1 to uuid ~3.3.2
```

File: package.json [hunk 3]
```diff
@@ -38,31 +38,31 @@
...
     "command-shell-lib": "1.0.0",
     "express": "~4.16.4",
     "jison": "0.4.18",
     "logops": "2.1.0",
-    "moment": "~2.22.2",
-    "moment-timezone": "~0.5.21",
-    "mongodb": "3.1.8",
-    "mongoose": "5.3.6",
+    "moment": "~2.24.0",
+    "moment-timezone": "^0.5.25",
```
Comment (fgalan) [inline package.json]: ^ -> ~, please

**Comment by fgalan** [comment]
IOTA Node Lib version 2.9.0 has been just released, so this PR gets "unblocked" and can progress. Please have a look to new comments and solve the conflicts on CHANGES_NEXT_RELEASE.

File: CHANGES_NEXT_RELEASE [hunk 4]
```diff
@@ -0,0 +1,2 @@
+- Upgrade from node-uuid ~1.4.1 to uuid ~3.3.2
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: This was only an example to show style. In fact, it is not correct this time as uuid didn't change in pacakges.json in this PR. Sorry for not being clearer...

I'll try to provide a real example for a dependency and another for dev dependency:

```
- Upgrade async dependency from 2.6.1 to 2.6.2
- Upgrade mocha dev dependency from 5.2.0 to 6.1.4
```

```
```

All changes in packages.json dependencies and dev dependencies should be included this way in CHANGES_NEXT_RELEASE.

File: package.json [hunk 5]
```diff
@@ -38,31 +38,31 @@
...
+    "query-string": "6.5.0",
     "request": "2.88.0",
     "revalidator": "~0.3.1",
-    "underscore": "~1.9.1"
+    "underscore": "~1.9.1",
+    "uuid": "~3.3.2"
   },
   "devDependencies": {
-    "coveralls": "~3.0.2",
+    "coveralls": "^3.0.3",
```
Comment (fgalan) [inline package.json]: ^ -> ~, pls

Comment (jason-fox) [reply package.json]: Fixed 6162b64

File: CHANGES_NEXT_RELEASE (hunk 4 above)
Comment (jason-fox) [reply CHANGES_NEXT_RELEASE]: Fixed 50476d3

**Comment by fgalan** [comment]
We prefer to include the ~ detail in the changelog. However, in order to simplify your work ;) I include bellow the new version of CNR with this taken into account. Just copy-paste it and upgrade the PR.

```
- Upgrade async dependency from 2.6.1 to 2.6.2
- Upgrade body-parser dependency from ~1.18.3 to ~1.19.0
- Upgrade moment dependency from ~2.22.2 to ~2.24.0
- Upgrade moment-timezone dependency from ~0.5.21 to ~0.5.25
- Upgrade mongodb dependency from 3.1.8 to 3.2.3
- Upgrade mongoose dependency from 5.3.6 to 5.5.4
- Upgrade query-string dependency from 6.2.0 to 6.5.0
- Upgrade coveralls dev dependency from ~3.0.2 to ~3.0.3
- Upgrade jshint dev dependency from ~2.9.6 to ~2.10.2
- Upgrade mocha dev dependency from 5.2.0 to 6.1.4
- Upgrade nock dev dependency from 10.0.1 to 10.0.6
- Upgrade timekeeper dev dependency from 2.1.2 to 2.2.0
- Set Nodejs 8 as minimum version in packages.json (effectively removing Nodev6 from supported versions) 
```

**Comment by jason-fox** [comment]
> - Upgrade async dependency from 2.6.1 to 2.6.2
> ... etc.
> - Set Nodejs 8 as minimum version in packages.json (effectively removing Nodev6 from supported versions)

Fixed 6cc8714

**Pull request merged**